Python/flowgraph.c

```diff
@@ -1384,6 +1393,66 @@ fold_tuple_on_constants(PyObject *const_cache,
     return SUCCESS;
 }
 
+// Replace LOAD_CONST x, LOAD_CONST y, LOAD_CONST z, BUILD_LIST 3
+// with BUILD_LIST 0, LOAD_CONST (x, y, z), LIST_EXTEND 1
+// or BUILD_SET & SET_UPDATE respectively.
+static int
+optimize_const_sequence(PyObject *const_cache,
+                        cfg_instr* inst,
+                        int n, PyObject *consts)
+{
+    assert(PyDict_CheckExact(const_cache));
+    assert(PyList_CheckExact(consts));
+    assert(inst[n].i_oparg == n);
+
+    int construct_list = inst[n].i_opcode == BUILD_LIST;
+    int build, extend;
+    if (!construct_list) {
+        assert(inst[n].i_opcode == BUILD_SET);
+        build = BUILD_SET;
+        extend = SET_UPDATE;
+    }
+    else {
+        build = BUILD_LIST;
+        extend = LIST_EXTEND;
+    }
```

Comment:
I find this easier to follow:

```suggestion
    int build = inst[n].i_opcode;
    assert(build == BUILD_LIST || build == BUILD_SET);
    int extend = (build == BUILD_LIST) ? SET_UPDATE : LIST_EXTEND
```